### PR TITLE
Add upgrade notes to cpu-management-policies

### PR DESCRIPTION
**Pull request by mac-chaffee**
See https://github.com/kubernetes-sigs/kubespray/issues/8535 for full context.

In short, if you merely change the `--cpu-manager-policy` flag from "none" to "static", kubelet will crashloop. You have to delete a specific file ~~and reboot the node~~. This PR documents that behavior: https://github.com/kubernetes/kubernetes/blob/v1.21.9/pkg/kubelet/cm/cpumanager/state/state_checkpoint.go#L58

~~Also: I'm using a newer version of hugo, which generates a .hugo_build.lock file. This is supposed to be .gitignored: https://gohugo.io/news/0.89.0-relnotes/#notes~~

### CODE REVIEW

**Review by sftim — DISMISSED** [review]
@mac-chaffee thanks for proposing these changes

Tthe change to `.gitignore` should not happen in the same commit as a revision to docs. Could you split these into separate commits or (ideally) into two different pull requests.

(Separate commits makes it easier to revert either change if needed, in isolation)

File: content/en/docs/tasks/administer-cluster/cpu-management-policies.md [hunk 1]
```diff
@@ -68,6 +68,23 @@ and `CPUManagerPolicyBetaOptions` feature gates. Diverging from the Kubernetes s
 feature gates guard groups of options, because it would have been too cumbersome to add a feature
 gate for each individual option.
 
+### Changing the CPU Manager Policy
+
+In order to change the CPU Manager policy on a node, perform the following steps:
+
+1. [Drain](/docs/tasks/administer-cluster/safely-drain-node) the node.
+2. Stop kubelet.
+3. Remove the old CPU manager state file `/var/lib/kubelet/cpu_manager_state`.
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/cpu-management-policies.md:77]:
```suggestion
3. Remove the old CPU manager state file. The path to this file is
   `/var/lib/kubelet/cpu_manager_state` by default.
```

File: content/en/docs/tasks/administer-cluster/cpu-management-policies.md [hunk 2]
```diff
@@ -68,6 +68,23 @@ and `CPUManagerPolicyBetaOptions` feature gates. Diverging from the Kubernetes s
...
+4. Edit the kubelet configuration to change the CPU manager policy to the desired value.
+4. Reboot the node.
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/cpu-management-policies.md:79]: It has to be a reboot? How come?

**Review by sftim — COMMENTED** [review]
/sig node

File: content/en/docs/tasks/administer-cluster/cpu-management-policies.md (hunk 2 above)
Comment (mac-chaffee) [reply content/en/docs/tasks/administer-cluster/cpu-management-policies.md:79]: I'm not sure of the specifics, but the KubeletConfiguration docs for changing `cpuManagerPolicy` tell you to do that: https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration

That advice came from this PR: https://github.com/kubernetes/kubernetes/pull/64354

But those docs are slated to be removed in 1.24: https://github.com/kubernetes/kubernetes/pull/106932/files#diff-e7cf3d9bacdf8ddf0db56a9a399993d2fc62897122206ecef17aee62c2e69d8fR342

Neither PR explicitly said why a reboot was necessary or if it still is. Just from looking at the PRs, I'm inclined to believe that a reboot _is_ still necessary (since mtaufen's PR was heavily reviewed, while updates to types.go in deprecation PR are likely to be skimmed-over).

If anyone in sig-node can clarify if a reboot is necessary for changing cpuManagerPolicy, that would be appreciated!

Comment (sftim) [reply content/en/docs/tasks/administer-cluster/cpu-management-policies.md:79]: I'm guessing that requesting a reboot is simpler to explain, but not actually needed. Ideally we'd document what you must actually restart.

For example, if you can:
```shell
systemctl stop kubelet.service
custom-cgroup-frob.sh # needs implementing - my point is that might be possible to do so
systemctl start kubelet.service
```
or
```shell
systemctl isolate emergency.target
# exit emergency environment
systemctl restart kubelet.target
```

then let's aim to document those steps (maybe by tracking a separate backlog issue).
Folks operating on bare metal may well value the extra detail.

Comment (klueska) [reply content/en/docs/tasks/administer-cluster/cpu-management-policies.md:79]: Reboot is not required — you can just remove the file `/var/lib/kubelet/cpu_manager_state` and restart your kubelet. This clears the state maintained by the CPUManager so that the cpu-sets set up by the new policy won’t conflict with it.

That said, if you do this, you may still have pods running that have cpu-sets set up for them by the old policy (and these may overlap with the cpu-sets that your new policy sets up).

If you are OK with this (or know this isn’t possible in your setup), then simply removing this file is OK.

File: .gitignore [hunk 3]
```diff
@@ -38,3 +38,4 @@ node_modules/
 # Generated files when building with make container-build
 .config/
 .npm/
+.hugo_build.lock
```
Comment (tengqm) [inline .gitignore:41]: as mentioned in the review comment, please move this file to a separate PR

File: content/en/docs/tasks/administer-cluster/cpu-management-policies.md [hunk 4]
```diff
@@ -46,7 +46,7 @@ management policies to determine some placement preferences on the node.
 ### Configuration
 
 The CPU Manager policy is set with the `--cpu-manager-policy` kubelet
-option. There are two supported policies:
+flag or the `cpuManagerPolicy` field in [KubeletConfiguration](/docs/reference/config-api/kubelet-config.v1beta1/). There are two supported policies:
```
Comment (tengqm) [inline content/en/docs/tasks/administer-cluster/cpu-management-policies.md:49]: Please avoid long lines like this by manually wrapping them. This will help the downstream localization efforts.

File: .gitignore (hunk 3 above)
Comment (mac-chaffee) [reply .gitignore]: Since the line doesn't do anything anyway, I just removed it. Figured it wasn't worth the extra overhead on either of us.

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-31721--kubernetes-io-main-staging.netlify.app/docs/tasks/administer-cluster/cpu-management-policies/#changing-the-cpu-manager-policy
